R/driver-sql-server.R

```diff
@@ -107,6 +107,30 @@ setMethod("dbExistsTable", c("Microsoft SQL Server", "SQL"),
   }
 )
 
+#' @details Call catalog-specific sp_tables to make sure we list the schemas in the
+#' appropriate database/catalog.
+#' @rdname SQLServer
+#' @usage NULL
+setMethod(
+  "odbcConnectionSchemas", "Microsoft SQL Server",
+  function(conn, catalog_name = NULL) {
+
+    if (is.null(catalog_name) || !nchar(catalog_name)) {
+      return(callNextMethod())
+    }
+    sproc <- paste(
+      dbQuoteIdentifier(conn, catalog_name),
+      dbQuoteIdentifier(conn, "dbo"),
+      dbQuoteIdentifier(conn, "sp_tables"), sep = ".")
```

Comment:
```suggestion
    sproc <- paste(
      dbQuoteIdentifier(conn, catalog_name),
      dbQuoteIdentifier(conn, "dbo"),
      dbQuoteIdentifier(conn, "sp_tables"), 
      sep = "."
    )
```

But do we need to quote `dbo` and `sp_tables`?